opentelemetry-sdk/src/export/logs/mod.rs

```diff
@@ -91,6 +92,25 @@ pub trait LogExporter: Send + Sync + Debug {
         // By default, all logs are enabled
         true
     }
+
+    /// This is a hint to ensure that the export of any Logs the exporter
+    /// has received prior to the call to this function SHOULD be completed
+    /// as soon as possible, preferably before returning from this method.
+    ///
+    /// This function SHOULD provide a way to let the caller know
+    /// whether it succeeded, failed or timed out.
+    ///
+    /// This function SHOULD only be called in cases where it is absolutely necessary,
+    /// such as when using some FaaS providers that may suspend the process after
+    /// an invocation, but before the exporter exports the completed logs.
+    ///
+    /// This function SHOULD complete or abort within some timeout. This function can be
+    /// implemented as a blocking API or an asynchronous API which notifies the caller via
+    /// a callback or an event. OpenTelemetry client authors can decide if they want to
+    /// make the flush timeout configurable.
+    fn force_flush(&mut self) -> BoxFuture<'static, ExportResult> {
```

Comment:
can we do normal method instead of async? 
`fn force_flush(&mut self) {}`

Reply:
I think this was discussed in community meeting, and there were suggestions to keep it async. cc @TommyCpp

Reply:
Specifically for scenario's where it is called from any async task, it will monopolizes the thread for the duration of that blocking operation, preventing other tasks to run on that thread.

Reply:
I understand we want to offer a async version of shutdown, but not sure if there is a need for force_flush. What scenario even requires user to call flush at all?

Reply:
>  What scenario even requires user to call flush at all?

It's part of the specs, so good to provide. Application can decide when to call - say for some high critical events when application want them to logged immediately, without waiting for the periodic flush. Also the specs talks about FaaS scenairo, where serverless function (acting as application) can get's suspended when there are no request, and which will also suspend the periodic flush of batch processor. The function may want to flush everything before getting suspended.

Reply:
makes sense!

Reply:
> What scenario even requires user to call flush at all?

I was thinking the same. We already have `force_flush` on `LoggerProvider`. That should work for users, right? Do we really need to add it for exporters as well? Is it just because the spec mentions it?

Reply:
Just to be clear - This is not to be called by user. But by `LoggerProvider::force_flush`.

`LoggerProvider::force_flush -> Processor::force_flush -> (for batch) exporter::export() for all events -> exporter::force_flush()`

I believe specs mentions it for a reason. We don't need it for OTLP exporters because of the way they are implemented. And also not required for the synchronous exporters like - etw and user_events. But we don't know how custom exporter are implemented - the successful export for them may be storing events initially to persistent storage. And this is then periodically flushed to some backend in background. This force_flush would acts as trigger for them to complete any such pending tasks if they want.

Reply:
Good point!